src/microsAuthentication/content.md

```diff
@@ -0,0 +1,1159 @@
+kind: ClusterRoleBinding
+metadata:
+  name: role-tokenreview-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+- kind: ServiceAccount
+  name: secret-store
+  namespace: secret-store
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  namespace: secret-store
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: secret-store
+  template:
+    metadata:
+      labels:
+        app: secret-store
+    spec:
+      serviceAccount: secret-store
+      containers:
+      - name: app
+        image: amitsaha/k8s-sa-volume-demo-ss:sa-1
+        env:
+        - name: LISTEN_ADDR
+          value: ":8081"
+        ports:
+        - containerPort: 8081
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: app
+  namespace: secret-store
+spec:
+  selector:
+    app: secret-store
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 8081
+```
+
+Compared to the API service, the secret store service requires a ClusterRoleBinding resource to be created which associates the secret-store service account to the `system:auth-delegator` ClusterRole.
+
+Go back to the terminal session where you deployed the secret store service from and run `kubectl --namepsace secret-store logs <pod id>` and you will see the following log lines where you can see the response from the TokenReview API call:
+
+```
+2020/08/21 09:09:10 v1.TokenReviewStatus{Authenticated:true,
+User:v1.UserInfo{Username:"system:serviceaccount:secret-store:secret-store",
+UID:"24ae6734-9ca1-44d8-af93-98b6e6f8da17", Groups:[]string{"system:serviceaccounts",
+"system:serviceaccounts:api", "system:authenticated"}, Extra:map[string]v1.ExtraValue(nil)},
+Audiences:[]string{"api"}, Error:""}
+```
+
+This is a Go structure version of the JSON response you saw earlier.
+
+Thus, you see how API service reads the service account token and passes it onto its request to the secret store service as a way to authenticate itself.
+
+The secret store service retrieves the token and then checks with the Kubernetes API whether it is valid or not.
+
+If valid, the secret store service allows the request from the API service to be processed.
+
+This suffers from two drawbacks:
+
+1. Anyone - human or otherwise with the permission to read the secret token in the api namespace can send a request to the secret store successfully. There is no embedded information in the token that the secret store can use to validate its intended audience.
+1. The secret token doesn't expire at all. If it is stolen, the only way to revoke it is to delete the service account and recreate it.
+
+You can solve the above problems by implementing solutions such as mutual TLS or using a JWT based solution with a central authority server.
+
+However, in Kubernetes, you can use the service account token volume projection feature to create time-bound and audience-specific service account tokens which do not persist in the cluster store.
```

Comment:
> which do not persist in the cluster store

Not sure what this means. the two features listed do not imply that the data is not stored?